arch/sparc/core/fault_trap.S

```diff
@@ -76,11 +85,19 @@ SECTION_FUNC(TEXT, __sparc_trap_fault)
 	st	%g1, [%sp + 96 + __z_arch_esf_t_y_OFFSET]
 
 	/* Enable traps, raise PIL to mask all maskable interrupts. */
-	or	%l0, PSR_PIL, %o0
-	wr	%o0, PSR_ET, %psr
+	or	%l0, PSR_PIL, %o2
+	wr	%o2, PSR_ET, %psr
 	nop
 	nop
 	nop
-	/* Exception stack frame prepared earlier is the first argument. */
-	call	_Fault
-	 add	%sp, 96, %o0
+	/*
+	 * reason is the first argument.
+	 * Exception stack frame prepared earlier is the second argument.
+	 */
+	call	z_sparc_fatal_error
```

Comment:
nit: seems like some whitespace issue here.

Reply:
The space before the instruction after the `call` is actually intended. It is part of the assembly convention to mark that the instruction is executing in a delay slot. For example after `call` or branch.